app/forms/flood_risk_engine/steps/user_type_form.rb

```diff
@@ -21,7 +21,13 @@ def params_key
       # when we save the enrollment
       def save
         model.org_type = org_type.to_sym
+        super
+        # Need to go back and then forward as organisation change will effect
+        # onward journey. Without the back and forth a user cannot go back
+        # through the user type selection.
+        enrollment.go_back
```

Comment:
this state engine appear to be very complex to work with, state management appears to be  expanding out from being encapsulated in the state machine itself to  lots of places where it's used. i.e as a developer not clear what this back/fwd stuff and double saved achieves, or under what circumstances I might need to do something similar